qemu/qmp.go

```diff
@@ -851,3 +852,23 @@ func (q *QMP) ExecSetMigrateArguments(ctx context.Context, url string) error {
 
 	return q.executeCommand(ctx, "migrate", args, nil)
 }
+
+// ExecHotplugMemory adds size of MB memory at specified slot to the guest
+func (q *QMP) ExecHotplugMemory(ctx context.Context, slot, size int) error {
+	args := map[string]interface{}{
+		"qom-type": "memory-backend-ram",
+		"id":       "mem" + strconv.Itoa(slot),
+		"props":    map[string]interface{}{"size": int64(size) << 20},
+	}
+	err := q.executeCommand(ctx, "object-add", args, nil)
+	if err != nil {
+		return err
+	}
+
+	args = map[string]interface{}{
+		"driver": "pc-dimm",
+		"id":     "dimm" + strconv.Itoa(slot),
+		"memdev": "mem" + strconv.Itoa(slot),
+	}
+	return q.executeCommand(ctx, "device_add", args, nil)
```

Comment:
We should probably attempt an object-del if this fails rather than returning straight away.